components/script/dom/bindings/str.rs

```diff
@@ -504,3 +550,26 @@ fn max_day_in_month(year_num: u32, month_num: u32) -> Result<u32, ()> {
         _ => Err(())
     }
 }
+
+// https://html.spec.whatwg.org/multipage/#week-number-of-the-last-day
+fn max_week_in_year(year: u32) -> u32 {
+    match Utc.ymd(year as i32, 1, 1).weekday() {
+        Weekday::Thu => 53,
+        Weekday::Wed => {
+            if is_leap_year(year) {
+                53
+            } else {
+                52
+            }
+        },
+        _ => 52
+    }
+}
+
+fn is_leap_year(year: u32) -> bool {
+    if year % 400 == 0 || (year % 4 == 0 && year % 100 != 0) {
```

Comment:
The `if` here is unnecessary, you can just return `year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)`.